crates/ruff/src/registry.rs

```diff
@@ -880,6 +881,18 @@ impl Rule {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DiagnosticKind {
+    /// The identifier of the corresponding [`Rule`].
+    pub rule: String,
+    /// The message body to display to the user, to explain the diagnostic.
+    pub body: String,
+    /// The message to display to the user, to explain the suggested fix.
+    pub commit: Option<String>,
```

Comment:
As an aside, this might be better-named `suggestion` or something. `commit` is bad (I think I came up with it).